Fix heredoc syntax error in AppImage build workflow

## Summary
The AppImage build job in the release workflow is failing due to improperly terminated heredoc strings. The closing delimiter 'EOF' is indented, which causes a shell syntax error.

## Steps to Reproduce
1. Trigger the release workflow
2. The build-appimages job executes
3. Job fails with error: "warning: here-document at line 23 delimited by end-of-file (wanted 'EOF')"

## Expected Behavior
The AppImage build should complete successfully without syntax errors.

## Actual Behavior
The job fails with a shell syntax error:
- Line 97: warning about unclosed heredoc
- Line 98: syntax error due to unexpected end of file

## Relevant Logs/Screenshots
From the failed workflow run: https://github.com/bitjungle/gopca/actions/runs/17289337452/job/49073169992

\`\`\`
/home/runema/actions-runner/_work/_temp/385da21b-6f37-46ab-9edd-d3f899a62af9.sh: line 97: warning: here-document at line 23 delimited by end-of-file (wanted 'EOF')
/home/runema/actions-runner/_work/_temp/385da21b-6f37-46ab-9edd-d3f899a62af9.sh: line 98: syntax error: unexpected end of file
Error: Process completed with exit code 2
\`\`\`

## Environment
- GitHub Actions self-hosted runner
- Shell: bash
- Workflow file: .github/workflows/release.yml

## Root Cause
In the release.yml file, there are two heredoc blocks used to create desktop files for the AppImages:

1. Lines 368-381: GoPCA desktop file heredoc
2. Lines 410-423: GoCSV desktop file heredoc  

Both have their closing 'EOF' delimiter indented with spaces, but heredoc closing delimiters must be at the beginning of the line with no leading whitespace.

## Proposed Solution
Remove the leading whitespace from the 'EOF' closing delimiters on lines 381 and 423 in .github/workflows/release.yml.

The fix is straightforward:
- Line 381: Change \`          EOF\` to \`EOF\`
- Line 423: Change \`          EOF\` to \`EOF\`

This will ensure the heredoc strings are properly terminated and the shell script can execute without syntax errors.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -366,18 +366,18 @@ jobs:
           
           # Create desktop file
           cat > appdir-gopca/gopca.desktop << 'EOF'
-          [Desktop Entry]
-          Version=1.0
-          Type=Application
-          Name=GoPCA
-          GenericName=PCA Analysis Tool
-          Comment=Professional Principal Component Analysis Tool
-          Exec=GoPCA %f
-          Icon=gopca
-          Terminal=false
-          Categories=Science;Math;Education;DataVisualization;
-          MimeType=text/csv;text/plain;
-          StartupNotify=true
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=GoPCA
+GenericName=PCA Analysis Tool
+Comment=Professional Principal Component Analysis Tool
+Exec=GoPCA %f
+Icon=gopca
+Terminal=false
+Categories=Science;Math;Education;DataVisualization;
+MimeType=text/csv;text/plain;
+StartupNotify=true
 EOF
           
           # Download icon from repository
@@ -408,18 +408,18 @@ EOF
           
           # Create desktop file
           cat > appdir-gocsv/gocsv.desktop << 'EOF'
-          [Desktop Entry]
-          Version=1.0
-          Type=Application
-          Name=GoCSV
-          GenericName=CSV Editor
-          Comment=Fast and Efficient CSV Data Editor
-          Exec=GoCSV %f
-          Icon=gocsv
-          Terminal=false
-          Categories=Office;Spreadsheet;Science;DataVisualization;
-          MimeType=text/csv;text/plain;text/tab-separated-values;
-          StartupNotify=true
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=GoCSV
+GenericName=CSV Editor
+Comment=Fast and Efficient CSV Data Editor
+Exec=GoCSV %f
+Icon=gocsv
+Terminal=false
+Categories=Office;Spreadsheet;Science;DataVisualization;
+MimeType=text/csv;text/plain;text/tab-separated-values;
+StartupNotify=true
 EOF
           
           # Download icon from repository

